### ISSUE TITLE
[BUG] "Nested Spawning" Attribute Corruption

### ISSUE THREAD

**Issue by jaborsh** [evennia/evennia#3686]
#### Describe the bug
Might be a long one to describe, so bear with me. TL;DR is at the bottom.

Imagine, if you will, prototype `WOMAN`. Within this prototype, we have the standard things: `{"key": "woman", "desc": "A woman."}` so on and so forth; however, you also have `{"spawn": {"clothing": [clothing_prototypes.ROBE]}}`.

In `at_object_post_spawn` you write some method to make sure that each prototype in the "spawn" attribute is created. It works. Woo!

Now you go about inspecting the robe, and all the attributes are exactly as they should be. After all, you've just created it. You call an update on the nested proto and everything still works well. How exciting!

Then you do something absolutely nefarious and modify the proto. You run batch_update and suddenly see all your mutable types are empty, not updating at all.

Specifically, the code that appears to be failing in this scenario is within `batch_update_objects_with_prototype`. Here is the code:
```py
obj.attributes.batch_add(
                            *(
                                (
                                    _init(akey, str),
                                    _init(aval, value_to_obj),
                                    acategory,
                                    alocks,
                                )
                                for akey, aval, acategory, alocks in val
                            )
                        )
 ```

The existence of akey, aval, acategory, and alocks lining up with our expectations has been verified; the mutable type is still present with all its associated information. Unfortunately, `_init(aval, value_to_obj)` fails and replaces it as empty.

TL:DR: When spawning a proto from one of another object's attributes, updating that proto does not work with mutable types.

#### To Reproduce
1. Create a prototype with an attribute that contains a list of prototypes. Nested prototypes need a mutable attribute.
2. Spawn the list of prototypes in `at_object_post_spawn`
3. Change nested proto mutable.
4. Attempt to update proto.
5. View empty mutables with examine or comparable method.

#### Expected behavior
Ideally proto mutables will function regardless of their spawn source.

#### Environment, Evennia version, OS etc
Evennia 4.4.1
Python 3.10.12

**Comment by InspectorCaracal** [comment]
I *do* still need to try to repro it, but it's on my list. >.>

**Comment by InspectorCaracal** [comment]
So, bad news, I cannot reproduce this in a minimal example, which means that most likely it is, as I suspected at the time, your code that's the issue.

Here's what I did:

INITIAL PROTOTYPE:
```python
SPECIAL_THING = {
  "key": "special thing",
  "stuff": {
    "a key": [1, 2, 3],
    "two keys": [4, 5, 6]
  },
}
```

Game logs:
```
> spawn special_thing
Spawned special thing.
> examine thing
------------------------------------------------------------------------------
Name/key: special thing (#3)
Typeclass: Object (typeclasses.objects.Object)
Location: Limbo (#2)
Home: Limbo (#2)
Locks:   call:true(); control:perm(Developer); delete:perm(Admin); drop:holds();
  edit:perm(Admin); examine:perm(Builder); get:all(); puppet:pperm(Developer);
  teleport:true(); teleport_here:true(); tell:perm(Admin); view:all()
Tags:   special_thing[from_prototype]
Persistent Attributes: 
  stuff={'a key': [1, 2, 3], 'two keys': [4, 5, 6]} [type: <class 'dict'>]
------------------------------------------------------------------------------
```
UPDATED PROTOTYPE:
```py
SPECIAL_THING = {
  "key": "special thing",
  "stuff": {
    "a key": [1, 2, 3],
    "NEFARIOUS CHANGES": ["a", "b", "c"],
  },
}
```
Game logs:
```
> reload
 Server restart initiated ...
 ... Server restarted.
> py from evennia.prototypes.spawner import batch_update_objects_with_prototype; objects = self.search("special thing", quiet=True); batch_update_objects_with_prototype("SPECIAL_THING", objects=objects)
>>> from evennia.prototypes.spawner import batch_update_objects_with_prototype; objects = self.search("special thing", quiet=True); batch_update_objects_with_prototype("SPECIAL_THING", objects=objects)
> examine thing
------------------------------------------------------------------------------
Name/key: special thing (#3)
Typeclass: Object (typeclasses.objects.Object)
Location: Limbo (#2)
Home: Limbo (#2)
Locks:   call:true(); control:perm(Developer); delete:perm(Admin); drop:holds();
  edit:perm(Admin); examine:perm(Builder); get:all(); puppet:pperm(Developer);
  teleport:true(); teleport_here:true(); tell:perm(Admin); view:all()
Tags:   special_thing[from_prototype]
Persistent Attributes: 
  stuff={'a key': [1, 2, 3], 'NEFARIOUS CHANGES': ['a', 'b', 'c']} [type: <class 'dict'>]
------------------------------------------------------------------------------
```

As you can see, everything updated as expected.
